compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -24,17 +24,15 @@ import dotc.transform.ValueClasses
  *       `ReplDriver#resetToInitial` is called, the accompanying instance of
  *       `Rendering` is no longer valid.
  */
-private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None):
+private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None,
+                              maxPrintElements: Int = 1000):
```

Comment:
I think it'd be better to not use a default value here and pass ctx.settings.XreplMaxPrintElements.value at all use-sites instead.